lib/phoenix_live_view/channel.ex

```diff
@@ -317,7 +321,7 @@ defmodule Phoenix.LiveView.Channel do
         end
       end)
 
-    new_socket = Utils.merge_flash(socket, flash)
+    new_socket = if redirected, do: Utils.merge_flash(socket, flash), else: socket
```

Comment:
Is this correct? If a component redirects, don't we want to merge its flash into the parent? 🤔

Reply:
if a component redirects, we definitely want to merge, no? If there is a flash to be added to the parent, you place it before redirecting and we have to maintain it or it's lost